float32 center_x
float32 center_y
float32 radius_x
float32 radius_y
float32 angle